docs/src/main/sphinx/connector/iceberg.md

```diff
@@ -1450,13 +1450,20 @@ it is being refreshed. Refreshing a materialized view also stores the
 snapshot-ids of all Iceberg tables that are part of the materialized view's
 query in the materialized view metadata. When the materialized view is queried,
 the snapshot-ids are used to check if the data in the storage table is up to
-date. If the data is outdated, the materialized view behaves like a normal view,
-and the data is queried directly from the base tables. Detecting outdated data
-is possible only when the materialized view uses Iceberg tables only, or when it
-uses a mix of Iceberg and non-Iceberg tables but some Iceberg tables are outdated.
-When the materialized view is based on non-Iceberg tables, querying it can
-return outdated data, since the connector has no information whether the
-underlying non-Iceberg tables have changed.
+date.
+
+The grace period of a materialized view defaults to infinity, and after each
+refresh the materialized view therefore behaves like a static table.
+
+If the data is outdated and the grace period of the materialized view is
+elapsed, the materialized view behaves like a normal view, and the data is
+queried directly from the base tables.
+
+Detecting outdated data is possible only when the materialized view uses Iceberg
+tables only, or when it uses a mix of Iceberg and non-Iceberg tables but some
+Iceberg tables are outdated. When the materialized view is based on non-Iceberg
+tables, querying it can return outdated data, since the connector has no
+information whether the underlying non-Iceberg tables have changed.
```

Comment:
When there are non-Iceberg tables, view is considered stale once GRACE elapses.

Reply:
Tried to pull this in with more rewording.